src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/ProjectHostProvider.cs

```diff
@@ -10,15 +10,15 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.LanguageServices
     internal sealed partial class ProjectHostProvider : IProjectHostProvider
     {
         [ImportingConstructor]
-        public ProjectHostProvider()
+        public ProjectHostProvider(UnconfiguredProject unconfiguredProject)
         {
-        }
+            Requires.NotNull(unconfiguredProject, nameof(unconfiguredProject));
 
-        public IUnconfiguredProjectHostObject GetUnconfiguredProjectHostObject(UnconfiguredProject unconfiguredProject)
-        {
-            return new UnconfiguredProjectHostObject(unconfiguredProject);
+            UnconfiguredProjectHostObject = new UnconfiguredProjectHostObject(unconfiguredProject);
         }
 
+        public IUnconfiguredProjectHostObject UnconfiguredProjectHostObject { get; }
+
         public IConfiguredProjectHostObject GetConfiguredProjectHostObject(IUnconfiguredProjectHostObject unconfiguredProjectHostObject, String projectDisplayName)
```

Comment:
>etConfiguredProjectHostObject(IUnconfiguredProjectHostObject unconfiguredProjectHostObject [](start = 45, length = 90)

This can be refactored too , since now we have IProjectHostProvider to live in UnconfiguredProject scope. Parameter unconfiguredProjectHostObject is not needed.